### PR TITLE
[prjtrellis] init

### PR DESCRIPTION
**Pull request by sjkelly**
This adds support for Prjtrellis, the open source bitstream documentation for ECP5 FPGAs.

### CODE REVIEW

File: P/Prjtrellis/build_tarballs.jl [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+dependencies = [
+    Dependency("Python_jll"),
+    Dependency("boost_jll"; compat="=1.76.0"), # max gcc7
+]
+
+# Bash recipe for building across all platforms
+script = raw"""
+cd prjtrellis/libtrellis
+cmake -DCMAKE_INSTALL_PREFIX=${prefix}
+make
```
Comment (giordano) [inline P/Prjtrellis/build_tarballs.jl:22]: This speeds up compilation quite a bit, especially on our build machine
```suggestion
make -j${nproc}
```

File: P/Prjtrellis/build_tarballs.jl [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+make install
+"""
+
+# These are the platforms we will build for by default, unless further
+# platforms are passed in on the command line
+platforms = filter!(p -> Sys.islinux(p), supported_platforms())
+platforms = expand_cxxstring_abis(platforms)
+# For some reason, building for CXX03 string ABI doesn't actually work, skip it
+filter!(x -> cxxstring_abi(x) != "cxx03", platforms)
```
Comment (giordano) [inline P/Prjtrellis/build_tarballs.jl:31]: Do you actually see this?

File: P/Prjtrellis/build_tarballs.jl [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+
+# The products that we will ensure are always built
+products = Product[
```
Comment (giordano) [inline P/Prjtrellis/build_tarballs.jl:34]:
```suggestion
products = [
```

File: P/Prjtrellis/build_tarballs.jl [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+name = "Prjtrellis"
+version = v"0.1.0"
+
+# Collection of sources required to complete build
+sources = [
+    GitSource("https://github.com/YosysHQ/prjtrellis.git", "dff1cbcb1bd30de7e96f8a059f2e19be1bb2e44d")
+]
+
+dependencies = [
+    Dependency("Python_jll"),
```
Comment (giordano) [inline P/Prjtrellis/build_tarballs.jl:14]: What is python used for?

File: P/Prjtrellis/build_tarballs.jl (hunk 1 above)
Comment (giordano) [inline P/Prjtrellis/build_tarballs.jl:22]:
```suggestion
make -j${nproc}
```

File: P/Prjtrellis/build_tarballs.jl (hunk 2 above)
Comment (giordano) [inline P/Prjtrellis/build_tarballs.jl:31]: Is this true?

File: P/Prjtrellis/build_tarballs.jl [hunk 5]
```diff
@@ -0,0 +1,46 @@
...
+    Dependency("boost_jll"; compat="=1.76.0"), # max gcc7
+]
+
+# Bash recipe for building across all platforms
+script = raw"""
+cd prjtrellis/libtrellis
+cmake -DCMAKE_INSTALL_PREFIX=${prefix}
```
Comment (giordano) [inline P/Prjtrellis/build_tarballs.jl:21]:
```suggestion
mkdir build && cd build
cmake -DCMAKE_INSTALL_PREFIX=${prefix} \
    -DCMAKE_TOOLCHAIN_FILE=${CMAKE_TARGET_TOOLCHAIN} \
    -DCMAKE_BUILD_TYPE=Release \
    ..
```

File: P/Prjtrellis/build_tarballs.jl (hunk 2 above)
Comment (sjkelly) [reply P/Prjtrellis/build_tarballs.jl:31]: Yes, I think it is related to Boost (again).

File: P/Prjtrellis/build_tarballs.jl (hunk 4 above)
Comment (sjkelly) [reply P/Prjtrellis/build_tarballs.jl:14]: There are python bindings via `pytrellis`. Some of the downstream packages (PRs incoming) use this for interface.

File: P/Prjtrellis/build_tarballs.jl (hunk 5 above)
Comment (sjkelly) [reply P/Prjtrellis/build_tarballs.jl:21]: Why add a `build` dir?

Comment (sjkelly) [reply P/Prjtrellis/build_tarballs.jl:21]: ah I see the `..` at the end.

Comment (giordano) [reply P/Prjtrellis/build_tarballs.jl]: When something fails and you want to restart from scratch, there is no way to do that with CMake unless you do a build in a subdirectory can you completely and safely delete

File: P/Prjtrellis/build_tarballs.jl (hunk 4 above)
Comment (giordano) [reply P/Prjtrellis/build_tarballs.jl:14]:
```suggestion
    Dependency("Python_jll"; compat="~3.8.1"),
```

**Pull request merged**